ui/src/App.tsx

```diff
@@ -138,101 +230,150 @@ const makeOffer = () => {
-              </div>
-            )}
-            {bldPurse && (
-              <div>
-                <b>BLD: </b>
-                {stringifyAmountValue(
-                  bldPurse.currentAmount,
-                  bldPurse.displayInfo.assetKind,
-                  bldPurse.displayInfo.decimalPlaces
+              <h4>Invitations:  </h4>
+              {
+                invites.length ? (
+                  <ul style={{ marginTop: 0, textAlign: 'left' }}>
+                    {invites.map(
+                      ({ description, customDetails }, index) => (
+                        <li key={index}>
+                          {description} {stringify(customDetails)}
+                        </li>
+                      )
+                    )}
+                  </ul>
+                ) : (
+                  'None'
                 )}
-              </div>
-            )}
-            {wallet && invitationPurse && (
-              <div>
-                <b>Invitations:  </b>
-                {
-                  invites.length ? (
-                    <ul style={{ marginTop: 0, textAlign: 'left' }}>
-                      {invites.map(
-                        ([name, number]) => (
-                          <li key={name}>
-                            {String(number)} {name}
-                          </li>
-                        )
-                      )}
-                    </ul>
-                  ) : (
-                    'None'
-                  )}
-              </div>
-            )}
-          </div>
+            </div>
+          )}
         </div>
-        <button onClick={onClick}>{buttonLabel}</button>
+      </div>
+      <div style={{ textAlign: 'left' }}>
+        <h4  >GIVE </h4>
+        <p>
+          {stringify(give) || 'Not yet provided'}
+        </p>
+        <h4 className="want" >WANT </h4>
+        <p>
+          {stringify(want) || 'Not yet provided'}
+        </p>
+        <br />
+        <label className="address" >Address:
+          <input
+            type="text"
+            value={recipient}
+            onChange={(e) => useForm.setState({ recipient: e.target.value })}
+            style={{ width: "30em" }}
+          />
+        </label>
+
+        <br />
+        {/* // select give asset
+        // fill give amount (with max button?)
+        // or drop down to select the notifier
+        // select want asset
+        // fill give amount
+        // or text field to past a description */}
+        <button onClick={() => setFirstOffer(useForm.getState())}>First Offer</button>
+        <button onClick={() => makeOffer(useForm.getState())}>Make Offer</button>
+        <br />
+        <button onClick={() => setMatchOffer(swaps[0])} disabled={!swaps.length}>Match First</button>
+        <button onClick={() => makeSecondOffer(useForm.getState())} disabled={!swaps.length}>Make Second Offer</button>
+      </div>
+    </div >
+  );
+};
+
+const App = () => {
+  useEffect(() => {
+    setup();
```

Comment:
Could this be renamed to `watchContract` or something? Trying to reason about, "what does `setup` do?"